### PR TITLE
Samples: Wifi: Added changes to help debugging

### PR DESCRIPTION
**Pull request by D-Triveni**
Log messages added to debug connection issues.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

### CODE REVIEW

File: samples/wifi/sta/src/main.c [hunk 1]
```diff
@@ -255,6 +285,18 @@ void main(void)
 	k_sleep(K_SECONDS(1));
 
 	wifi_connect();
-	k_sleep(K_SECONDS(30));
+
+	k_thread_start(client_status);
+	k_thread_join(client_status, K_SECONDS(30));
```
Comment (rlubos) [inline samples/wifi/sta/src/main.c:290]: Why to create a separate thread only to blocking the current one, waiting for the new thread to terminate? Couldn't `cmd_wifi_status()` be printed directly from main?

File: samples/wifi/sta/src/main.c [hunk 2]
```diff
@@ -53,38 +57,50 @@ static int cmd_wifi_status(void)
...
 	}
+	return 0;
+}
 
+static void status_thread(void *p1, void *p2, void *p3)
+{
+	int i;
 
-	return 0;
+	for (i = 0; i < CONFIG_CONNECTION_TIMEOUT_MS; i++) {
```
Comment (rlubos) [inline samples/wifi/sta/src/main.c:94]: `CONNECTION_TIMEOUT_MS` - the config name seems confusing, currently it's used as an interation limit, not the timeout value.

File: samples/wifi/sta/src/main.c (hunk 1 above)
Comment (D-Triveni) [reply samples/wifi/sta/src/main.c:290]: @rlubos Yes, `cmd_wifi_status()` can be printed from main. But, for readbility, we are polling the status in new thread.

File: samples/wifi/sta/Kconfig [hunk 3]
```diff
@@ -8,6 +8,18 @@ source "Kconfig.zephyr"
 
 menu "Nordic Sta sample"
 
+config CLIENT_STATUS_THREAD_STACK_SIZE
```
Comment (krish2718) [inline samples/wifi/sta/Kconfig:11]: this is a STA sample so, may be use STA as prefix instead of CLIENT or no prefix at all for some?

File: samples/wifi/sta/src/main.c (hunk 1 above)
Comment (krish2718) [reply samples/wifi/sta/src/main.c:290]: As the connect is a non-blocking API, we can use main thread itself. Also, the connect starts a status thread anyway, so, if we can just log a couple of debug prints that can be enabled by customer, no need for polling status here, just can add a timeout based on connected.

File: samples/wifi/sta/src/main.c [hunk 4]
```diff
@@ -255,6 +285,18 @@ void main(void)
 	k_sleep(K_SECONDS(1));
 
 	wifi_connect();
-	k_sleep(K_SECONDS(30));
+
+	k_thread_start(client_status);
+	k_thread_join(client_status, K_SECONDS(30));
+
+	if (context.connected) {
+		if (k_sem_take(&sem_dhcp, K_SECONDS(70)) == -EAGAIN) {
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:293]: should be a Kconfig or at least a define based on DHCP timings

**Review by krish2718 — CHANGES_REQUESTED** [review]
Please split changes in to mulitples commits (logging, functional1, functional2 etc)

**Review by richabp — APPROVED** [review]

**Comment by D-Triveni** [comment]
Addressed all previous comments.
Sta connection logging functionality moved to main thread.

File: samples/wifi/sta/prj.conf [hunk 5]
```diff
@@ -69,7 +73,7 @@ CONFIG_ENTROPY_GENERATOR=y
 
 # Logging
 CONFIG_LOG=y
-CONFIG_PRINTK=y
+CONFIG_LOG_BUFFER_SIZE=2048
```
[the remark this answers is not in the document]
Comment (D-Triveni) [reply samples/wifi/sta/prj.conf:76]: Tried reducing the buffer size to 1024, with that, few messages are getting dropped. So, it's defined as 2048.

File: samples/wifi/sta/Kconfig [hunk 6]
```diff
@@ -8,6 +8,14 @@ source "Kconfig.zephyr"
 
 menu "Nordic Sta sample"
 
+config MAXIMUM_ITERATIONS
```
Comment (krish2718) [inline samples/wifi/sta/Kconfig:11]: This doesn't look like a user facing entry, its more of an internal state. This should either be renamed (connection_timeout) or removed from Kconfig.

File: samples/wifi/sta/Kconfig [hunk 7]
```diff
@@ -8,6 +8,14 @@ source "Kconfig.zephyr"
...
+	int "Number of iterations for logging the station connection status"
+	default 100
+
+config STATUS_POLLING_MS
```
Comment (krish2718) [inline samples/wifi/sta/Kconfig:15]: Even this is not something user understands or cares, this was mainly for our debugging.

File: samples/wifi/sta/src/main.c [hunk 8]
```diff
@@ -255,6 +263,23 @@ void main(void)
...
+		k_sleep(K_MSEC(CONFIG_STATUS_POLLING_MS));
+		cmd_wifi_status();
+		if (context.connect_result) {
+			break;
+		}
+	}
+
+	if (context.connected) {
+		if (k_sem_take(&sem_dhcp, K_SECONDS(DHCP_TIMEOUT)) == -EAGAIN) {
+			LOG_ERR("DHCP Connection Timed Out");
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:277]:
```suggestion
			LOG_ERR("DHCP handshake timed out");
```

File: samples/wifi/sta/src/main.c [hunk 9]
```diff
@@ -53,35 +58,33 @@ static int cmd_wifi_status(void)
 
 	if (net_mgmt(NET_REQUEST_WIFI_IFACE_STATUS, iface, &status,
 				sizeof(struct wifi_iface_status))) {
-		printk("Status request failed\n");
+		LOG_INF("Status request failed");
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:61]: can you please separate these logging changes to a separate commit.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: samples/wifi/sta/src/main.c [hunk 10]
```diff
@@ -31,14 +31,21 @@ LOG_MODULE_REGISTER(sta, CONFIG_LOG_DEFAULT_LEVEL);
 				NET_EVENT_WIFI_DISCONNECT_RESULT)
 
 #define MAX_SSID_LEN        32
+#define DHCP_TIMEOUT        70
+#define MAXIMUM_ITERATIONS  100
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:35]: This is too generic, can we just use `CONN_TIMEOUT_S` instead?

File: samples/wifi/sta/src/main.c [hunk 11]
```diff
@@ -255,6 +264,23 @@ void main(void)
...
+
+	if (context.connected) {
+		if (k_sem_take(&sem_dhcp, K_SECONDS(DHCP_TIMEOUT)) == -EAGAIN) {
+			LOG_ERR("DHCP handshake timed out");
+		}
+		k_sleep(K_SECONDS(30));
+	} else if (!context.connect_result) {
+		LOG_ERR("Connection Timed Out");
+	}
 	wifi_disconnect();
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:284]: This be done only in success case.

**Review by krish2718 — COMMENTED** [review]
I don't see the commit with logging changes?

**Comment by D-Triveni** [comment]
> I don't see the commit with logging changes?
I've pushed another commit.

**Review by krish2718 — APPROVED** [review]
For the logging commit log add a line about increased buffer size.

File: samples/wifi/sta/src/main.c [hunk 12]
```diff
@@ -255,6 +264,22 @@ void main(void)
...
+		if (context.connect_result) {
+			break;
+		}
+	}
+
+	if (context.connected) {
+		if (k_sem_take(&sem_dhcp, K_SECONDS(DHCP_TIMEOUT)) == -EAGAIN) {
+			LOG_ERR("DHCP handshake timed out");
+		}
+		k_sleep(K_SECONDS(30));
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:280]: We could make this also configurable, CONNECTION_IDLE_TIME

**Review by rlubos — APPROVED** [review]

**Comment by peknis** [comment]
No changelog entry?
